woocommerce/woocommerce-ios
# Order Notes: Fixes classification logic

This PR fixes the order note classification logic (customer, private, system) on the order details screen:

![note_fix](https://user-images.githubusercontent.com/154014/51197509-612e0b00-18b7-11e9-8213-a33b6c0f5874.png)

The issue here was in the note data model — any note that had`"customer_note": true` and `"author": "system"` was marked as a system note because that was inspected first.

Fixing this required adjusting the logic in `OrderNoteTableViewCell.swift` to check if the note is a "customer note" _before_ checking the author. So the resulting truth table looks like so:

| Customer Note  | Author | Label Applied in app |
| ------------- | ------------- | ------------- |
| true  | (doesn't matter) | Customer Note |
| false  | system | System Note |
| false  | not system | Private Note |

Fixes: #604 

## Testing

1. Build and run the unit tests - verify they are ✅ 
2. Build and run the app — log into the same store as in #604
3. Open the order details for all the problematic orders in #604
4. Verify the order notes are classified in the same way as WC Android does it

----

@astralbodies would you mind giving this PR a quick 👀 ? Thanks!!
